src/cluster/peers.rs

```diff
@@ -0,0 +1,125 @@
+use crate::net::{Host, ToHosts};
+use logos::{Lexer, Logos};
+
+#[derive(Logos, Debug, PartialEq)]
+enum Token {
+    #[token("[")]
+    OpenBracket,
+
+    #[token("]")]
+    CloseBracket,
+
+    #[regex("[0-9a-zA-Z-./_:]+")]
+    Text,
+
+    #[error]
+    #[regex(r"[ \t\f,]+", logos::skip)]
+    Error,
+}
+
+fn parse_error(lex: &Lexer<Token>, source: &str) -> String {
+    format!(
+        "Failed to parse peers: {}, at {:?} ({})",
+        source,
+        lex.span(),
+        lex.slice()
+    )
+}
+
+pub fn parse_peers_info(info_peers: &str) -> Result<Vec<Host>> {
+    let mut lex = Token::lexer(info_peers);
+
+    let _peer_gen = match lex.next() {
+        Some(Token::Text) => lex.slice(),
+        _ => bail!(ErrorKind::BadResponse(parse_error(&lex, info_peers))),
+    };
+    let default_port_str = match lex.next() {
+        Some(Token::Text) => lex.slice(),
+        _ => bail!(ErrorKind::BadResponse(parse_error(&lex, info_peers))),
+    };
+
+    let default_port = match default_port_str.parse::<u16>() {
+        Ok(port) => port,
+        Err(_) => bail!(ErrorKind::BadResponse(format!(
+            "Invalid default port: {}",
+            default_port_str
+        ))),
+    };
+
+    match lex.next() {
+        Some(Token::OpenBracket) => parse_peers(info_peers, &mut lex, default_port),
+        _ => Ok(Vec::new()),
+    }
+}
+
+fn parse_peers(info_peers: &str, lex: &mut Lexer<Token>, default_port: u16) -> Result<Vec<Host>> {
+    let mut peers = Vec::new();
+    loop {
+        match lex.next() {
+            Some(Token::OpenBracket) => peers.extend(parse_peer(info_peers, lex, default_port)?),
+            Some(Token::CloseBracket) => return Ok(peers),
+            _ => bail!(ErrorKind::BadResponse(parse_error(&lex, info_peers))),
+        }
+        lex.next(); // Close brackets
+    }
+}
+
+fn parse_peer(info_peers: &str, lex: &mut Lexer<Token>, default_port: u16) -> Result<Vec<Host>> {
+    let _id = match lex.next() {
+        Some(Token::Text) => lex.slice(),
+        _ => bail!(ErrorKind::BadResponse(parse_error(&lex, info_peers))),
+    };
+    let _tls_hostname = match lex.next() {
+        Some(Token::Text) => lex.slice(),
+        _ => bail!(ErrorKind::BadResponse(parse_error(&lex, info_peers))),
+    };
+    match lex.next() {
+        Some(Token::OpenBracket) => (),
+        _ => bail!(ErrorKind::BadResponse(parse_error(&lex, info_peers))),
+    };
+
+    let hosts = match lex.next() {
+        Some(Token::Text) => lex.slice(),
+        _ => bail!(ErrorKind::BadResponse(parse_error(&lex, info_peers))),
+    }
+    .to_hosts_with_default_port(default_port)?;
+
+    lex.next(); // Close brackets
+    Ok(hosts)
+}
+
+#[cfg(test)]
+mod tests {
+    use std::vec;
+
+    use super::*;
+
+    #[test]
+    fn parse_peers_works() {
+        let work = "6,3000,[[12A0,aerospike.preprod.crto.in,[192.168.1.13:4333]],[11A1,aerospike.preprod.crto.in,[192.168.3.78]],[11A1,aerospike.preprod.crto.in,[localhost]]]";
```

Comment:
The used Hostnames and IPs look like real ones. Maybe change that to dummy addresses?

Reply:
IP were dummy but not the hostname indeed.

Changed them also corrected a bug where the tls_hostname is ommited